SignalServiceKit/src/Messages/Interactions/TSMessage.m

```diff
@@ -226,33 +227,64 @@ - (BOOL)hasAttachments
 
 - (NSString *)debugDescription
 {
-    if ([self hasAttachments]) {
+    if ([self hasAttachments] && self.body.length > 0) {
+        NSString *attachmentId = self.attachmentIds[0];
+        return [NSString
+            stringWithFormat:@"Media Message with attachmentId: %@ and caption: '%@'", attachmentId, self.body];
+    } else if ([self hasAttachments]) {
         NSString *attachmentId = self.attachmentIds[0];
         return [NSString stringWithFormat:@"Media Message with attachmentId:%@", attachmentId];
     } else {
         return [NSString stringWithFormat:@"%@ with body: %@", [self class], self.body];
     }
 }
 
+// TODO: This method contains view-specific logic and probably belongs in NotificationsManager, not in SSK.
 - (NSString *)previewTextWithTransaction:(YapDatabaseReadTransaction *)transaction
 {
```

Comment:
These changes LGTM, but do you want to duplicate them into the `description` method too?

Reply:
Done.